.github/workflows/fedora-rpm.yml

```
@@ -0,0 +1,47 @@
+name: Fedora RPM
+
+on:
+  push:
+    branches: [ "master" ]
+  pull_request:
+    branches: [ "master" ]
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.number && format('pr{0}', github.event.number) || github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  Fedora:
+    name: Build on Fedora
+    runs-on: ubuntu-latest
+    
+    steps:
+      - name: Setup Podman
+        run: |
+          sudo apt update
+          sudo apt-get -y install podman
+          podman pull fedora:40
+
+      - name: Get source
+        uses: actions/checkout@v3
+        with:
+          path: 'miracle-wm'
+
+      - name: Build the rpm
+        run: |
+          {
+              echo 'FROM fedora:40'
+              echo '# set TZ to ensure the test using timestamp'
+              echo 'RUN dnf -y update'
+              echo 'RUN dnf install -y rpmdevtools rpmlint dnf-plugins-core'
+              echo 'RUN dnf clean all'
+              echo 'RUN rpmdev-setuptree'
+              echo 'COPY miracle-wm miracle-wm-0.1.0'
+              echo 'RUN chmod -R a+rwx miracle-wm-0.1.0'
+              echo 'RUN tar --create --file miracle-wm-0.1.0.tar.gz miracle-wm-0.1.0'
+              echo 'RUN mv miracle-wm-0.1.0.tar.gz ~/rpmbuild/SOURCES'
+              echo 'RUN cp ./miracle-wm-0.1.0/rpm/miracle-wm.spec ~/rpmbuild/SPECS'
+              echo 'RUN dnf -y builddep ~/rpmbuild/SPECS/miracle-wm.spec'
+              echo 'RUN rpmbuild -bs ~/rpmbuild/SPECS/miracle-wm.spec' 
```

Maybe do this instead?

```suggestion
              echo 'FROM fedora:rawhide'
              echo '# set TZ to ensure the test using timestamp'
              echo 'RUN dnf install -y mock mock-core-configs distribution-gpg-keys'
              echo 'COPY miracle-wm miracle-wm-0.1.0'
              echo 'RUN chmod -R a+rwx miracle-wm-0.1.0'
              echo 'RUN tar --create --file miracle-wm-0.1.0.tar.gz miracle-wm-0.1.0'
              echo 'RUN mock --isolation=simple --root=fedora-40-x86_64 --sources . --spec miracle-wm-0.1.0/rpm/miracle-wm.spec' 
```

This mimics how the package is built in Fedora itself.

---

Let's try it out!

---

Eh, no dice. Not going to spend any more time on the CI though for now. The main thing here was feedback on the package request :)